rules_gapic/bazel/src/main/java/com/google/api/codegen/bazel/Buildozer.java

```diff
@@ -0,0 +1,161 @@
+package com.google.api.codegen.bazel;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Buildozer {
+  private static Buildozer instance = null;
+  private File tempdir = null;
+  private File buildozerBinary = null;
+  private List<String> batch = null;
+
+  private Buildozer() throws IOException {
+    tempdir = Files.createTempDirectory("build_file_generator_").toFile();
+    final String resourcePath = "/rules_gapic/bazel/buildozer.bin";
```

Comment:
As we discussed offline plese try using `classpath_resoruce` in java_binary, maybe it will let your avoid copying the binary from java jar resoruce.

Reply:
Actually, I just figured that regular `data` still works.